src/helper/remote_config/client.cpp

```diff
@@ -32,10 +28,10 @@ config_path config_path::from_path(const std::string &path)
     return config_path{base_match[3].str(), base_match[2].str()};
 }
 
-client::client(std::unique_ptr<http_api> &&arg_api, service_identifier sid,
-    remote_config::settings settings, const std::vector<product> &products)
-    : api_(std::move(arg_api)), id_(dds::generate_random_uuid()),
-      sid_(std::move(sid)), settings_(std::move(settings))
+client::client(std::unique_ptr<http_api> &&arg_api, service_identifier &&sid,
```

Comment:
Why change this at all? Passing by value is superior to passing by rvalue-reference if you're always going to be moving and copying, only pass by rvalue-reference when you only want to allow moves.